inference-engine/src/mkldnn_plugin/memory_descs/cpu_memory_desc_utils.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cpu_memory_desc.h"
+#include "memory_descs/cpu_memory_desc_utils.h"
+#include "mkldnn_memory.h"
+#include "memory_descs/onednn_blocked_memory_desc.h"
+#include "utils/general_utils.h"
+#include "utils/cpu_utils.hpp"
+#include <limits>
+#include <vector>
+#include <numeric>
+#include <blob_factory.hpp>
+#include <dnnl_types.h>
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+namespace MKLDNNPlugin {
+
+MemoryDescPtr MemoryDescUtils::makeDescriptor(const mkldnn::memory::desc &desc) {
+    if (desc.data.format_kind == dnnl_blocked) {
+        return std::unique_ptr<OnednnBlockedMemoryDesc>(new OnednnBlockedMemoryDesc(desc));
+    } else {
+        return MKLDNNPlugin::make_unique<MKLDNNMemoryDesc>(desc);
+    }
+}
+
+std::unique_ptr<MKLDNNMemoryDesc> MemoryDescUtils::convertToMKLDNNMemoryDesc(const MemoryDesc& desc) {
+    if (MemoryDescType::CpuBlocked == desc.getType()) {
+        return convertToMKLDNNMemoryDesc(*(desc.as<CpuBlockedMemoryDesc>()));
+    } else if (MemoryDescType::Mkldnn & desc.getType()) {
+        return MKLDNNPlugin::make_unique<MKLDNNMemoryDesc>(desc.as<MKLDNNMemoryDesc>()->getMklDesc());
+    } else {
+        IE_THROW() << "Cannot convert MemoryDesc to MKLDNNMemoryDesc";
+    }
+}
+
+std::unique_ptr<MKLDNNMemoryDesc> MemoryDescUtils::convertToMKLDNNMemoryDesc(const CpuBlockedMemoryDesc& desc) {
+    return std::unique_ptr<OnednnBlockedMemoryDesc>(new OnednnBlockedMemoryDesc(desc.getPrecision(), desc.getShape(), desc.getBlockDims(),
+                                                              desc.getOrder(), desc.getOffsetPadding(), desc.getOffsetPaddingToData(), desc.getStrides()));
+}
+
+std::unique_ptr<OnednnBlockedMemoryDesc> MemoryDescUtils::convertToOnednnBlockedMemoryDesc(const InferenceEngine::TensorDesc& desc) {
+    const auto &blkDesc = desc.getBlockingDesc();
+    return std::unique_ptr<OnednnBlockedMemoryDesc>(new OnednnBlockedMemoryDesc(desc.getPrecision(), Shape(desc.getDims()), blkDesc.getBlockDims(),
+                                                                                blkDesc.getOrder(), blkDesc.getOffsetPadding(),
+                                                                                blkDesc.getOffsetPaddingToData(), blkDesc.getStrides()));
+}
+
+MemoryDescPtr MemoryDescUtils::applyUndefinedOffset(const MemoryDesc& desc) {
+    if (desc.getType() == MemoryDescType::Mkldnn) {
+        return desc.clone();
```

Comment:
i guess better leave as is, to avoid inserting additional check that ```desc_type != mkldnn```
in all scenario we anyway need to initialize config even if it is not possible to apply undefined offset